gcore/gdaldrivermanager.cpp

```diff
@@ -766,6 +766,120 @@ char **GDALDriverManager::GetSearchPaths(const char *pszGDAL_DRIVER_PATH)
     return papszSearchPaths;
 }
 
+/************************************************************************/
+/*                          LoadPlugin()                                */
+/************************************************************************/
+
+/**
+ * \brief Load a single GDAL driver/plugin from shared libraries.
+ *
+ * This function will load a single named driver/plugin from shared libraries.
+ * It searches the "driver path" for .so (or .dll) files named
+ * "gdal_{name}.[so|dll|dylib]" or "ogr_{name}.[so|dll|dylib]", then tries to
+ * call a function within them called GDALRegister_{name}(), or failing that
+ * called GDALRegisterMe().
+ * 
+ * \see GDALDriverManager::AutoLoadDrivers() for the rules used to determine
+ * which paths are searched for plugin library files.
+ */
+
+CPLErr GDALDriverManager::LoadPlugin(const char *name)
+{
+#ifdef GDAL_NO_AUTOLOAD
```

Comment:
Potentially we could remove the #ifdef GDAL_NO_AUTOLOAD (if doing so, the comment in GDALRegisterPlugin() should be removed too). It is useful in the  GDALDriverManager::AutoLoadDrivers() context, since that function is called by GDALAllRegister(), and thus may load "random" code without it being obvious.
LoadPlugin() is a manual action, so users know what they do.